src/base/bittorrent/session.cpp

```diff
@@ -1347,6 +1361,48 @@ bool Session::addTorrent_impl(AddTorrentData addData, const MagnetUri &magnetUri
     return true;
 }
 
+bool Session::findIncompleteFiles(TorrentInfo &torrentInfo, QString &savePath) const
+{
+    auto findInDir = [](const QString &dirPath, TorrentInfo &torrentInfo) -> bool
+    {
+        bool found = false;
+        if (torrentInfo.filesCount() == 1) {
+            const QString filePath = dirPath + torrentInfo.filePath(0);
+            if (QFile(filePath).exists()) {
+                found = true;
+            }
+            else if (QFile(filePath + QB_EXT).exists()) {
+                found = true;
+                torrentInfo.renameFile(0, torrentInfo.filePath(0) + QB_EXT);
+            }
+        }
+        else {
+            QStringList allFiles =
+                    findAllFiles(dirPath + Utils::String::fromStdString(torrentInfo.nativeInfo()->files().name()));
+            for (int i = 0; i < torrentInfo.filesCount(); ++i) {
```

Comment:
OK. Then:
a) why does it continue looping if `true` was already set?
b) what if the torrent first file is, say, "description.txt" or "info.nfo" or any other common name which is easy to find at a random place?